soh/soh/Enhancements/randomizer/draw.cpp

```diff
@@ -275,5 +275,28 @@ extern "C" void Randomizer_DrawTriforcePieceGI(PlayState* play, GetItemEntry get
     } else if (current == required && triforcePieceScale > 0.00008f) {
         gSPDisplayList(POLY_XLU_DISP++, (Gfx*)gTriforcePieceCompletedDL);
     }
+
+    CLOSE_DISPS(play->state.gfxCtx);
+}
+
+extern "C" void Randomizer_DrawMysteryItem(PlayState* play, GetItemEntry getItemEntry) {
+    Color_RGB8 color = { 0, 60, 100 };
+    if (CVarGetInteger(CVAR_COSMETIC("World.MysteryItem.Changed"), 0)) {
+        color = CVarGetColor24(CVAR_COSMETIC("World.MysteryItem.Value"), color);
+    }
+
+    OPEN_DISPS(play->state.gfxCtx);
+
+    Gfx_SetupDL_25Xlu(play->state.gfxCtx);
+
+    gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(play->state.gfxCtx, (char*)__FILE__, __LINE__), G_MTX_MODELVIEW | G_MTX_LOAD);
```

Comment:
```suggestion
    gSPMatrix(POLY_XLU_DISP++, MATRIX_NEWMTX(play->state.gfxCtx), G_MTX_MODELVIEW | G_MTX_LOAD);
```

Reply:
Following review on Discord, this will require a change in macros.h that is probably outside the purview of this PR, so I'll go ahead and mark this as resolved.